mne/io/fiff/tests/test_raw.py

```diff
@@ -888,9 +888,16 @@ def test_hilbert():
     picks = picks_meg[:4]
 
     raw2 = raw.copy()
+    raw3 = raw.copy()
+    raw4 = raw.copy()
     raw.apply_hilbert(picks)
     raw2.apply_hilbert(picks, envelope=True, n_jobs=2)
 
+    # Test custom N
+    raw3.apply_hilbert(picks, N=raw3.n_times + 100)
+    assert_equal(raw3._data.shape, raw._data.shape)
```

Comment:
Hmmm, ok so it does seem like the padding will affect the signal in more than just the peri-zero region. Does anyone in the MNE world have a more fundamental understanding of how padding and FFTs interact with each other?

e.g.:
```python
out1 = sg.hilbert(raw._data[0, :1000])
out2 = sg.hilbert(raw._data[0, :1000], N=1500)

print(raw3._data[:5, 0])
print(raw._data[:5, 0])
```
Output:
```python
[ -2.58713276e-12 +1.87655655e-11j   1.31091416e-12 -9.42899838e-12j
  -1.45324088e-14 -1.43578007e-13j  -2.77344749e-12 +2.45830856e-11j
   5.02734681e-13 +0.00000000e+00j]
[ -2.58713270e-12   1.31091411e-12  -1.45324088e-14  -2.77344739e-12
   5.02734681e-13]
```